src/cregistry/entry.c

```diff
@@ -344,14 +325,14 @@ int reg_entry_delete(reg_entry* entry, reg_error* errPtr) {
     if (ports) {
         sqlite3_finalize(ports);
     }
-    if (files) {
-        sqlite3_finalize(files);
+    if (*pFiles) {
+        sqlite3_finalize(*pFiles);
     }
-    if (dependencies) {
-        sqlite3_finalize(dependencies);
+    if (*pDependencies) {
+        sqlite3_finalize(*pDependencies);
     }
-    if (portgroups) {
-        sqlite3_finalize(portgroups);
+    if (*pPortgroups) {
+        sqlite3_finalize(*pPortgroups);
```

Comment:
I'd replace those with a loop, too – less chance for missing the free if we ever wanted to add an additional query.